lib/src/bitarray.dart

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection' show ListMixin;
+import 'dart:typed_data' show Uint32List;
+import 'package:collection/collection.dart' show NonGrowableListMixin;
+
+/// A BitArray implementation for Dart to store boolean values.
+///
+/// Uses list of integers as internal storage to reduce memory usage.
+/// Implemented as fixed-length [List] of [bool].
+/// Throws error when calling any operation with length modifying.
+///
+/// Support several ways to perform logical operations (NOT, OR, AND, XOR):
+/// * instance methods (mutate [this]): [BitArray.not], [BitArray.or], [BitArray.and], [BitArray.xor];
+/// * producing new instance of [BitArray]:
+///     * logical operators: [BitArray.~], [BitArray.|], [BitArray.&], [BitArray.^].
+///     * logical gates (as static methods): [BitArray.notGate], [BitArray.orGate], [BitArray.andGate], [BitArray.xorGate].
+/// All arguments for not-unary logical operations must have the same [length] or [ArgumentError] will be thrown.
+class BitArray with ListMixin<bool>, NonGrowableListMixin<bool> {
```

Comment:
If keeping it as a list, call it "BitList' or `BoolList`. We don't use "Array" anywhere else.

This is both an implementation and an interface (it introduces new methods on top of `List`).
As an implementation class, it's fine to refer to the implementation strategy (`Bit`) rather than the API (`bool`). A reusable interface should not be mentioning implementation details.
So, let's assume that this is not a reusable interface, there are not expected to be other implementations than this one. Then I'd call it `BitList`.

Reply:
It becomes an (slightly) reusable interface when someone implements the growable version.

Reply:
Will be renamed to BitList.